test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsUtilityTests.cs

```diff
@@ -316,5 +316,66 @@ public void GetDefaultPushSource_WithNullSettings_Throws()
             ex.Should().NotBeNull();
             ex.Should().BeOfType<ArgumentNullException>();
         }
+
+        [Fact]
+        public void GetNamespaceMode_WithNullSettings_Throws()
+        {
+            var ex = Record.Exception(() => SettingsUtility.GetNamespaceMode(settings: null));
+
+            ex.Should().NotBeNull();
+            ex.Should().BeOfType<ArgumentNullException>();
+        }
+
+        [Fact]
+        public void GetNamespaceMode_WithValidKeyValue_Success()
+        {
+            // Arrange
+            var nugetConfigPath = "NuGet.Config";
+            var config = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <config>
+        <add key=""namespaceMode"" value=""singleSourcePerPackage"" />
+    </config>
+</configuration>";
+
+            using var mockBaseDirectory = TestDirectory.Create();
+
+            SettingsTestUtils.CreateConfigurationFile(nugetConfigPath, mockBaseDirectory, config);
+            var settings = new Settings(mockBaseDirectory);
+
+            // Act
+            var result = SettingsUtility.GetNamespaceMode(settings);
+
+            // Assert
+            Assert.Equal(NamespaceMode.SingleSourcePerPackage, result);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(null)]
+        [InlineData("bad value")]
+        [InlineData("atLeastOneSourcePerPackage")]
+        public void GetNamespaceMode_WithInValidOrDefaultValueReturnsDefaultMode_Success(string namespaceMode)
+        {
+            // Arrange
+            var nugetConfigPath = "NuGet.Config";
+            var config = $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <config>
+        <add key=""namespaceMode"" value= ""{namespaceMode}"" />
+    </config>
+</configuration>";
+
+            using var mockBaseDirectory = TestDirectory.Create();
+
+            SettingsTestUtils.CreateConfigurationFile(nugetConfigPath, mockBaseDirectory, config);
+            var settings = new Settings(mockBaseDirectory);
+
+            // Act
+            var result = SettingsUtility.GetNamespaceMode(settings);
+
+            // Assert
+            Assert.Equal(NamespaceMode.AtLeastOneSourcePerPackage, result);
```

Comment:
Wouldn't we want config file loading to fail, so customers know that their typo isn't silently giving them the wrong behaviour.  Imagine they set `SingleSourcPerPackage` (note there's one character deleted from the correct value). If they're expecting to be protected from multiple sources per package, but it's silently ignored, then namespace configuration errors won't be reported.

Reply:
Great point Andy. This logic is inspired by the existing implementation for `SignatureValidationMode` which returns `Accept` by default. 
https://github.com/NuGet/NuGet.Client/blob/cb290a427545a9214f22d9d6196c9e3c81380611/src/NuGet.Core/NuGet.Configuration/Utility/SettingsUtility.cs#L90-L102

I will change `GetNamespaceMode` logic to throw an exception if `namespaceMode` setting in the `NuGet.Config` file is invalid.

Reply:
Addressed this feedback in the latest commit. Please review again.